napari/layers/surface/surface.py

```diff
@@ -0,0 +1,309 @@
+from typing import Union
+from xml.etree.ElementTree import Element
+import numpy as np
+from copy import copy
+import vispy.color
+from ..base import Layer
+from ...util.event import Event
+from ..image._constants import Rendering, Interpolation, AVAILABLE_COLORMAPS
+from ...util.status_messages import format_float
+from ...util.misc import calc_data_range, increment_unnamed_colormap
+from vispy.color import get_color_names, Color
+
+
+class Surface(Layer):
+    """
+    Surface layer renders meshes onto the canvas.
+
+    Parameters
+    ----------
+    data : (N, D) array
+        Vertices of mesh triangles.
+    faces : (M, 3) array of int
+        Indices of mesh triangles.
+    values : (N,) array
+        Values used to color vertices.
```

Comment:
As above: either `vertices, faces, values`, or `data` should contain the three elements.

Reply:
hmm, i'll give it some thought ...

Reply:
on the one hand i like having `data` the standard across all our layers, but on the other hand I like `data` only having an array inside it as a tuple might be a bit weird to find, i'm not sure right now :-(

Reply:
Yep. I don't think having it always be a tuple is good, it will make a lot of code more complicated. It should be a tuple when it makes sense.

Reply:
we now have the three separate attributes